### PR TITLE
kotatogram-desktop.tg_owt: unpin openssl_1_1

### PR DESCRIPTION
**Pull request by VergeDX**
This reverts commit f83a8af.

###### Description of changes

https://github.com/NixOS/nixpkgs/pull/229392#issuecomment-1564547894

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/networking/instant-messengers/telegram/kotatogram-desktop/tg_owt.nix [hunk 1]
```diff
@@ -66,6 +88,9 @@ stdenv.mkDerivation {
       --replace '"libGL.so.1"' '"${libGL}/lib/libGL.so.1"' \
       --replace '"libgbm.so.1"' '"${mesa}/lib/libgbm.so.1"' \
       --replace '"libdrm.so.2"' '"${libdrm}/lib/libdrm.so.2"'
+
+    rm -r src/third_party/libsrtp
+    cp -r --no-preserve=mode ${libsrtp} $_
```
Comment (NickCao) [inline pkgs/applications/networking/instant-messengers/telegram/kotatogram-desktop/tg_owt.nix:93]:
```suggestion
    cp -r --no-preserve=mode ${libsrtp} src/third_party/libsrtp
```

**Comment by ilya-fedin** [comment]
@VergeDX have you checked group video chats work? If yes then I'd trust you as I have weak hardware and building kotatogram consumes multiple hours for me.

**Comment by ilya-fedin** [comment]
Nice!

**Comment by NickCao** [comment]
@ofborg build kotatogram-desktop

**Review by NickCao — APPROVED** [review]

**Comment by ilya-fedin** [comment]
I'm currently examining the failure of kotatogram-desktop for ZHF, looks like this PR has broke build on x86_64-darwin with the following log: https://hydra.nixos.org/build/224147055/nixlog/5

**Comment by ilya-fedin** [comment]
The failure is quite weird given that there's a patch deleting the reference to crypto/cipher/cipher_test_cases.c. It feels like patching doesn't happen for darwin what would be really weird...

**Comment by ilya-fedin** [comment]
Oh, wait, looks like mis-read it, it adds the reference to it. Looks like the problem is with postPatch step that is applied only to Linux.

**Pull request merged**